src/statement_generator.cpp

```diff
@@ -517,9 +517,17 @@ unique_ptr<TableRef> StatementGenerator::GenerateSubqueryRef() {
 }
 
 unique_ptr<TableRef> StatementGenerator::GenerateTableFunctionRef() {
-	auto function = make_uniq<TableFunctionRef>();
-	auto &table_function_ref = Choose(generator_context->table_functions);
-	auto &entry = table_function_ref.get().Cast<TableFunctionCatalogEntry>();
+	auto original_val = generator_context->table_functions.size();
+	auto random_val = RandomValue(original_val);
+	auto table_function_ref = &generator_context->table_functions[random_val];
+	while (table_function_ref->get().type == CatalogType::TABLE_MACRO_ENTRY) {
+		random_val += 1 % original_val;
+		if (random_val == original_val) {
```

Comment:
sorry, but realizing this is still wrong.

original_val is set to the size of the `generator_context->table_functions` vector. 
original_val should be the first random value (i.e the `RandomValue()` returned on line 521.

Then `random_val` can be increased in the while loop (with the modulo) until it is equal to `original_val`

Reply:
Basically it should look like
```c++

auto num_table_functions = generator_context->table_functions.size();
auto random_val = RandomValue(original_val);
auto original_val = random_val;
auto table_function_ref = &generator_context->table_functions[random_val];
while (table_function_ref->get().type == CatalogType::TABLE_MACRO_ENTRY) {
    random_val += 1; // I also think the + and modulo needs to be split up, 
                     // otherwise you get random_val = random_val + (1 % num_table_functions)
    random_val = random_val % num_table_functions;
    if (random_val == original_val) {
        ....
    }
}
```

Reply:
Sorry, I didn't get all these details earlier. Fixed